app/models/room.rb

```diff
@@ -1,3 +1,14 @@
 class Room < ApplicationRecord
   belongs_to :hotel
+  has_many :guest_rooms
+  has_many :guests, through: :guest_rooms
+
+  def add_guest_to_room(guest_id)
+    guest = Guest.find(guest_id)
+    self.guests << guest
+  end
+
+  def count_guests
+    self.guests.count
```

Comment:
Excellent encapsulation! It's usually helpful to also include `distinct` in here (although it wasn't explicit in the user story that it needed to be, so we accept either case!)